ddtrace/baggage/baggage.go

```diff
@@ -37,6 +38,8 @@ func Set(ctx context.Context, key, value string) context.Context {
 	if !ok {
```

Comment:
nit: Probably, we could use the same for the Remove.
```suggestion
	if !ok || bm == nil {
```

Reply:
good catch